### ISSUE TITLE
Support Hex equal

### ISSUE THREAD

**Issue by SilentCicero** [EthWorks/Waffle#369]
One very easy and common feature Waffle should include in it's base Chai matchers is a hex equals.

This will match hex values forcing lowerCase.

Example:

```
expect(ethers.utils.hexZeroPad('0xaaBB')).to.be.hex(ethers.utils.hexZeroPad('0xaabb'));
```

These two values are essentially the same, and thus using a special `hex` equal, we can easily compare. This can be used to compare things like addresses to daw data, like create2 situations for example.

**Comment by marekkirejczyk** [comment]
I think it is a good idea, would you like to prepare a PR?

**Comment by marekkirejczyk** [comment]
I assume zero in front can be ignored, so that following will succeed:
```ts
expect('0xaaBB').to.be.hex(ethers.utils.hexZeroPad('0xaabb'));
```

Otherwise we are just talking about case insensitive comparision.

**Comment by rzadp** [comment]
@marekkirejczyk or @sz-piotr This issue can be closed now I think.

**Comment by marekkirejczyk** [comment]
Released in [3.2.2](https://github.com/EthWorks/Waffle/releases/tag/3.2.2)
